python/distributed-ucxx/distributed_ucxx/distributed_patches.py

```diff
@@ -0,0 +1,41 @@
+from distributed import Scheduler, Worker
+from distributed.utils import log_errors
+
+import ucxx
+
+_scheduler_close = Scheduler.close
+_worker_close = Worker.close
+
+
+def _stop_notifier_thread_and_progress_tasks():
+    ucxx.stop_notifier_thread()
+    ucxx.core._get_ctx().progress_tasks.clear()
```

Comment:
This presumably means we can't have multiple schedulers running simultaneously (probably this is a dask restriction too). Since the context is global and not tied to a scheduler.

Reply:
TBH, I didn't even think of a scenario where we would have multiple schedulers. In any case, I'm not really sure how we could handle this in a better way than how it's currently written, so anything slightly less common (like multiple schedulers) will demand some further evaluation. Perhaps some sort of reference counting would be useful, but given the unpredictability of garbage collection in Python I'm not sure if that would work either.

Reply:
No, I think this is fine.